holoviews/element/raster.py

```diff
@@ -408,45 +211,157 @@ class Image(SheetCoordinateSystem, Raster):
+        """
+        Ensures old-style unpickled Image types without an interface
+        use the ImageInterface.
+        """
+        self.__dict__ = state
+        if isinstance(self.data, np.ndarray):
+            self.interface = ImageInterface
+        super(Dataset, self).__setstate__(state)
+
+
+    def aggregate(self, dimensions=None, function=None, spreadfn=None, **kwargs):
+        agg = super(Image, self).aggregate(dimensions, function, spreadfn, **kwargs)
+        return Curve(agg) if isinstance(agg, Dataset) and len(self.vdims) == 1 else agg
+
+
+    def select(self, selection_specs=None, **selection):
+        """
+        Allows selecting data by the slices, sets and scalar values
+        along a particular dimension. The indices should be supplied as
+        keywords mapping between the selected dimension and
+        value. Additionally selection_specs (taking the form of a list
+        of type.group.label strings, types or functions) may be
+        supplied, which will ensure the selection is only applied if the
+        specs match the selected object.
+        """
+        if selection_specs and not any(self.matches(sp) for sp in selection_specs):
+            return self
 
+        selection = {self.get_dimension(k).name: slice(*sel) if isinstance(sel, tuple) else sel
+                     for k, sel in selection.items() if k in self.kdims}
+        coords = tuple(selection[kd.name] if kd.name in selection else slice(None)
+                       for kd in self.kdims)
 
-    def _convert_element(self, data):
-        if isinstance(data, (Raster, HeatMap)):
-            return data.data
+        indexed = False
+        if not any([isinstance(el, slice) for el in coords]):
+            indexed = True
+            selection = {kd.name: c for kd, c in zip(self.kdims, self.closest(coords))}
+
+        # Compute new bounds
+        shape = self.interface.shape(self)
+        ys, xs = shape[:2]
+        xidx, yidx = coords
+        xdim, ydim = self.kdims
+        l, b, r, t = self.bounds.lbrt()
+        xdensity, ydensity = self.xdensity, self.ydensity
+        xunit = (1./xdensity)
+        yunit = (1./ydensity)
+        if isinstance(xidx, slice):
+            l = l if xidx.start is None else max(l, xidx.start)
+            r = r if xidx.stop is None else min(r, xidx.stop)
+        if isinstance(yidx, slice):
+            b = b if yidx.start is None else max(b, yidx.start)
+            t = t if yidx.stop is None else min(t, yidx.stop)
+        bounds = BoundingBox(points=((l, b), (r, t)))
+
+        # Apply new bounds
+        slc = Slice(bounds, self)
+
+        # Apply scalar and list indices
+        l, b, r, t = slc.compute_bounds(self).lbrt()
+        if indexed:
+            pass
+        elif not isinstance(xidx, slice):
+            if not isinstance(xidx, (list, set)): xidx = [xidx]
+            if len(xidx) > 1:
+                xdensity = xdensity*(float(len(xidx))/xs)
+            idxs = []
```

Comment:
Ah, yes, leftover from when computing the bounds and slicing the data were still combined.